content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md

```diff
@@ -0,0 +1,161 @@
+  for a Pod.
+
+Parameters for ResourceClass and ResourceClaim are stored in separate objects,
+typically using the type defined by a {{< glossary_tooltip
+term_id="CustomResourceDefinition" text="CRDs" >}} that was created when
+installing a resource driver.
+
+The `core/v1` `PodSpec` defines ResourceClaims that are needed for a Pod in a new
+`resourceClaims` field. Entries in that list reference either a ResourceClaim
+or a ResourceClaimTemplate. When referencing a ResourceClaim, all Pods using
+this PodSpec (for example, inside a Deployment or StatefulSet) share the same
+ResourceClaim instance. When referencing a ResourceClaimTemplate, each Pod gets
+its own instance.
+
+The `ResourceRequirements` for a container define whether a container gets
+access to these resource instances, which makes it possible to share resources
+between one or more containers.
+
+## Scheduling
+
+In contrast to native resources (CPU, RAM) and extended resources (managed by a
+device plugin, advertised by kubelet), the scheduler has no knowledge of what
+dynamic resources are available in a cluster or how they could be split up to
+satisfy the requirements of a specific ResourceClaim. Resource drivers are
+responsible for that. They mark ResourceClaims as "allocated" once resources
+for it are reserved. This also then tells the scheduler where in the cluster a
+ResourceClaim is available.
+
+ResourceClaims can get allocated as soon as they are created ("immediate
+allocation"), without consider which Pods will use them. The default is to
+delay allocation until a Pod gets scheduled which needs the ResourceClaim
+("wait for first consumer").
+
+In that mode, the scheduler checks all ResourceClaims needed by a Pod and
+creates a PodScheduling object where it informs the resource drivers
+responsible for those ResourceClaims about nodes that the scheduler considers
+suitable for the Pod. The resource drivers respond by excluding nodes that
+don't have enough of the driver's resources left. Once the scheduler has that
+information, it selects one node and stores that choice in the PodScheduling
+object. The resource drivers then allocate their ResourceClaims so that the
+resources will be available on that node. Once that is complete, the Pod
+gets scheduled.
+
+As part of this process, ResourceClaims also get reserved for the
+Pod. Currently ResourceClaims can either be used exclusively by a single Pod or
+an unlimited number of Pods.
+
+If allocation fails despite the earlier availability check, for example because
+of resource contention, or there isn't any node which has all required
+resources, the Pod cannot get scheduled. This avoids the scenario where a Pod
+gets scheduled onto one node and then cannot run there, which is bad because
+such a pending Pod also blocks all other resources like RAM or CPU that were
+set aside for it. There is a recovery mechanism that avoids deadlocks by
+freeing already allocated and unused ResourceClaims.
```

Comment:
Who is this documentation for? Does the end user need to know all of these details?

Reply:
It's for anyone who wants to get a quick overview of this feature without reading the KEP, which could be an end user or a hardware vendor. I think explaining how PodScheduling is used is relevant, because users or admins might encounter it when troubleshooting something related to Pods with ResourceClaims.

Pointing out that Pods do not get scheduled unless resources are available IMHO is important. The part about the recovery mechanism can be left out.